### ISSUE TITLE
Compilation error during conan create . 

### ISSUE THREAD

**Issue by KAITO-FBX** [NTNU-IHB/FMI4cpp#61]
Thank you for providing this package.    
Some error messages (Cf later in the post) during compilation after the command :
conan create . FMI4cpp/testing while I am in the /Documents/FMI/FMI4cpp-conan$ directory.

My current system is the following :
Linux 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux
I also add to manually install cmake to this version as the Debian distro doesn't provide it.
cmake version 3.14.2

Could you be so kind as to providing some advices for me to solve this issue?
Thank you for your help.

.....

Here is one of the error messages ....
/ModelDescription.hpp:29:20: fatal error: optional: No such file or directory
 #include <optional>
....
Following is a detailed description of the error message ending with the previous lines.
[ 31%] Building CXX object src/CMakeFiles/fmi4cpp.dir/fmi4cpp/fmi2/import/fmi2Fmu.cpp.o
[ 36%] Building CXX object src/CMakeFiles/fmi4cpp.dir/fmi4cpp/fmi2/import/fmi2CoSimulationLibrary.cpp.o
[ 42%] Building CXX object src/CMakeFiles/fmi4cpp.dir/fmi4cpp/fmi2/import/fmi2Library.cpp.o
In file included from /home/frederic/.conan/data/FMI4cpp/0.6.1/FMI4cpp/testing/build/2ddc3fb6da02007e31c4d4885b23776a2dab579a/FMI4cpp/src/../include/fmi4cpp/fmi2/fmi2.hpp:29:0,
                 from /home/frederic/.conan/data/FMI4cpp/0.6.1/FMI4cpp/testing/build/2ddc3fb6da02007e31c4d4885b23776a2dab579a/FMI4cpp/src/../include/fmi4cpp/fmi4cpp.hpp:28,
                 from /home/frederic/.conan/data/FMI4cpp/0.6.1/FMI4cpp/testing/build/2ddc3fb6da02007e31c4d4885b23776a2dab579a/FMI4cpp/src/../include/fmi4cpp/common/driver/fmu_driver.hpp:31,
                 from /home/frederic/.conan/data/FMI4cpp/0.6.1/FMI4cpp/testing/build/2ddc3fb6da02007e31c4d4885b23776a2dab579a/FMI4cpp/src/fmi4cpp/common/driver/fmu_driver.cpp:29:
/home/frederic/.conan/data/FMI4cpp/0.6.1/FMI4cpp/testing/build/2ddc3fb6da02007e31c4d4885b23776a2dab579a/FMI4cpp/src/../include/fmi4cpp/fmi2/xml/ModelDescription.hpp:29:20: fatal error: optional: No such file or directory
 #include <optional>
                    ^

**Comment by KAITO-FBX** [comment]
After some research I have found that some compilers do not provide <optional> but rather <experimental/optional>   

Should I upgrade my compiler as well ?
Mine is currently
gcc version 6.3.0 20170516 (Debian 6.3.0-18+deb9u1) 

Thank you

**Comment by markaren** [comment]
FMI4cpp is quite aggressive in terms of requirements (c++17, cmake, boost). This is mostly due to the fact that I can't (won't) support more configuration than I use myself (Ubuntu 18.04, windows 10).  

I will try to update the conan package to target 0.7.0 (not quite finished yet). Perhaps I should replace experimental/optional with boost/optional in that release, since boost is a dependency anyway..

**Comment by KAITO-FBX** [comment]
Thank you for your answer and I completely understand and agree with your position.   
I'll try to see what I can do on my side with the upgrades.
Great work by the way :-)

**Comment by markaren** [comment]
I've updated https://github.com/markaren/FMI4cpp-conan to use 0.7.0-RC2. Could you try it?

**Comment by KAITO-FBX** [comment]
Thank you 
I just pulled the FMI4cpp-conan and ran the install command after a :"conan remove FMI4cpp"
The previous error has indeed disappeared and it seems that we have through the process.
I still need some time to go around and try to use the code but it seems fine so far.
I also joined a file with the install message for your verification.

Thank you very much for your help.

[FMI4cpp-intall.txt](https://github.com/NTNU-IHB/FMI4cpp/files/3084337/FMI4cpp-intall.txt)
